### PR TITLE
Add compatibility with HTTP CORS requests

### PR DESCRIPTION
**Pull request by Stratehm**
Corresponding Bugzilla: https://bugs.eclipse.org/bugs/show_bug.cgi?id=478037

Signed-off-by: Antoine Besnard <stratehm@hotmail.com>

### CODE REVIEW

File: bundles/io/org.eclipse.smarthome.io.rest/src/main/java/org/eclipse/smarthome/io/rest/internal/filter/CorsFilter.java [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+/**
+ * A PostMatching filter used to add CORS HTTP headers on responses for requests with CORS
+ * headers.
+ *
+ * Based on http://www.w3.org/TR/cors
+ *
+ * This implementation does not allow specific request/response headers nor cookies (allowCredentials).
+ *
+ * @author Antoine Besnard
```
Comment (dnobel) [inline bundles/io/org.eclipse.smarthome.io.rest/src/main/java/org/eclipse/smarthome/io/rest/internal/filter/CorsFilter.java]: Please change to something like "@author Antoine Besnard - Initial contribution"

File: bundles/io/org.eclipse.smarthome.io.rest.test/src/test/groovy/org/eclipse/smarthome/io/rest/test/filter/CorsFilterTest.groovy [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+import javax.ws.rs.core.MultivaluedHashMap
+import javax.ws.rs.core.MultivaluedMap
+
+import org.eclipse.smarthome.io.rest.internal.filter.CorsFilter
+import org.junit.Test
+
+/**
+ * Test for the {@link CorsFilter} filter.
+ *
+ * @author Antoine Besnard
```
Comment (dnobel) [inline bundles/io/org.eclipse.smarthome.io.rest.test/src/test/groovy/org/eclipse/smarthome/io/rest/test/filter/CorsFilterTest.groovy]: Please change to : "@author Antoine Besnard - Initial contribution"

**Comment by dnobel** [comment]
Thanks for the PR, looks really great. Just two minor comments in the JavaDoc. Usually we add an information what was modified by the author.

Besides that one question: By now this PR enables CORS by default for any application that will use Eclipse SmartHome. Do you see any disadvantage or possible security risk, so that someone else don´t want to have CORS enabled? Questions is there a need to make it configurable in order to deactivate it? Or would you say it´s in general good to have CORS enabled and there is no additional risk?

**Comment by Stratehm** [comment]
You are right. It should be disabled by default to avoid security risk (like a third party website which takes controls of your items). One can never be too careful with security. People should wittingly enable such features.

**Comment by kaikreuzer** [comment]
So maybe we should add a configuration policy "required" to the OSGi service, so that it is only registered at all, if there is a configuration (that again should be checked for a setting like cors=enabled or so). Would that make sense?

**Comment by Stratehm** [comment]
Looks OK to me.

**Comment by kaikreuzer** [comment]
So could you add that to this PR, please?

**Comment by Stratehm** [comment]
Yes, I will :)

**Comment by Stratehm** [comment]
I have tried to use the "required" configuration policy without success. The smarthome.cfg file looks not to be read by the ConfigurationAdmin and so the service is never activated.

Is the solution used in the last commit good enough ?

**Comment by kaikreuzer** [comment]
> Is the solution used in the last commit good enough ?

Not really, but if you like, simply squash your commits and I will try to add the necessary adaptions as a separate commit from myself when merging this PR.

**Comment by Stratehm** [comment]
> Not really, but if you like, simply squash your commits and I will try to add the necessary adaptions as a separate commit from myself when merging this PR.

It will indeed be better. Thanks

Squashed.

**Comment by kaikreuzer** [comment]
Done, thanks!

**Pull request merged**